github/src/main/java/io/syndesis/git/GitWorkflow.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.git;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.RemoteAddCommand;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.api.errors.NoFilepatternException;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.transport.PushResult;
+import org.eclipse.jgit.transport.URIish;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.syndesis.core.SyndesisServerException;
+
+public class GitWorkflow {
+
+    private static final Logger log = LoggerFactory.getLogger(GitWorkflow.class);
+    /**
+     * Commits and pushes the files to the Git Repository
+     * 
+     * @param repo- an existing github repository
+     * @param message- commit message
+     * @param files- map of file paths along with their content
+     * @param credentials- Git credentials, for example username/password, authToken, personal access token
+     * @return RevCommit, the commit info
+     */
+    public RevCommit createFiles(String repoHtmlUrl, String repoName, String message, Map<String, byte[]> files,
+            UsernamePasswordCredentialsProvider credentials) {
+
+        try {
+            // create temporary directory
+            Path workingDir = Files.createTempDirectory(repoName);
```

Comment:
Not really an issue. just thought about performance implications because of the layered container file system (which might have other issues) but agree that we should keep simple.

just asking ;-)